src/VisualStudio/Core/Def/Implementation/ChangeSignature/AddParameterDialogViewModel.cs

```diff
@@ -25,9 +25,9 @@ internal class AddParameterDialogViewModel : AbstractNotifyPropertyChanged
 
         private readonly SemanticModel _semanticModel;
 
-        public AddParameterDialogViewModel(Document document, int insertPosition)
+        public AddParameterDialogViewModel(Document document, int insertPosition, INotificationService notificationService)
         {
-            _notificationService = document.Project.Solution.Workspace.Services.GetService<INotificationService>();
```

Comment:
❕ This is by definition correct. `INotificationService` is a MEF part. If you have a copy of it, it is only possible for that copy to be correct.